scapy/fields.py

```diff
@@ -460,7 +464,10 @@ def getfield(self, pkt, s):
             l = self.length_from(pkt)
         elif self.count_from is not None:
             c = self.count_from(pkt)
-            
+        elif not isinstance(self.next_cls_cb, type(None)):
```

Comment:
I applied the `is not None` modification.
I use `elif` because `self.next_cls_cb` is exclusive, just as `self.count_from` and `self.length_from` are mutually exclusive. 
In fact, `self.next_cls_cb` is not that much exclusive. It is more like it is flexible enough to implement the equivalent of the two others inside the callback. You can implement `count_from` using a "memory" (as demonstrated in the unit tests), and the "length limit" can be emulated by either counting the number of remaining bytes or by adding the successive serialized versions of parsed packets. I can add a unit test demonstrating the length limit implementation, if you would like.